### PR TITLE
[constants][docs] Add missing descriptions for few properties under `NativeConstants`

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Follow up https://github.com/expo/expo/pull/30021

# How

Expo Constants API reference has some missing descriptions for properties under `NativeConstants`. This PR adds missing descriptions for: `debugMode`, `executionEnvironment`, `isHeadless`, and `plarform` (Could only add missing descriptions for the properties I understood).

It also fixes formatting issues to avoid using Latin abbreviations and use bold text for file names as per our writing style guide.

Changes only applied to `unversioned` API reference.

# Test Plan

By running yarn build & et gdad -p expo-constants to update the property description. Also, manually running docs to confirm changes in generated docs file: http://localhost:3002/versions/unversioned/sdk/constants/#nativeconstants.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-constants/src/Constants.types.ts [hunk 1]
```diff
@@ -134,6 +137,9 @@ export interface NativeConstants {
    * @deprecated Moved to `expo-device` as [`Device.deviceYearClass`](./device/#deviceyearclass).
    */
   deviceYearClass: number | null;
+  /**
+   * Returns the current execution environment. Current values are `bare`, `standalone` and `storeClient`.
```
Comment (wschurman) [inline packages/expo-constants/src/Constants.types.ts:141]: optional: potential types seem self-documenting enough but it's up to you

```suggestion
   * Returns the current execution environment.
```

File: packages/expo-constants/src/Constants.types.ts [hunk 2]
```diff
@@ -199,6 +208,9 @@ export interface NativeConstants {
    * @hidden
    */
   supportedExpoSdks?: string[];
+  /**
+   * Returns the platform-specific manifest object.
+   */
```
Comment (wschurman) [inline packages/expo-constants/src/Constants.types.ts:213]:
```suggestion
  /**
   * Returns the platform-specific platform manifest object. 
   */
```

(somehow we should probably communicate that this is an independent concept from `manifest`/`manifest2` manifests, no idea how to do that cleanly)

**Review by wschurman — APPROVED** [review]

File: packages/expo-constants/src/Constants.types.ts (hunk 1 above)
Comment (amandeepmittal) [reply packages/expo-constants/src/Constants.types.ts:141]: Done.

File: packages/expo-constants/src/Constants.types.ts (hunk 2 above)
Comment (amandeepmittal) [reply packages/expo-constants/src/Constants.types.ts]: @wschurman, based on your feedback, I've updated it to the following and also added a callout to note the point you mentioned:

```ts
  /**
   * Returns the specific platform manifest object.
   *
   * > **Note**: This is distinct from the `manifest` and `manifest2`.
   */
```

Let me know if you've any further feedback.

**Pull request merged**